hydrolib/core/io/net/models.py

```diff
@@ -0,0 +1,916 @@
+from __future__ import annotations
+
+from datetime import datetime
+from pathlib import Path
+from platform import uname
+from typing import List, Optional, Union, Dict
+from _pytest.mark import structures
+
+import meshkernel as mk
+from meshkernel.py_structures import GeometryList
+import netCDF4 as nc
+import numpy as np
+from numpy.lib import poly
+from pydantic import Field
+from pydantic.types import PyObject
+from pydantic.utils import ValueItems
+
+from hydrolib.core import __version__
+from hydrolib.core.basemodel import BaseModel
+from hydrolib.core.io.net.reader import UgridReader
+from hydrolib.core.io.net.writer import UgridWriter
+
+import matplotlib
+from matplotlib.collections import LineCollection
+
+
+def split_by(gl: mk.GeometryList, by: float) -> list:
+    """Function to split mk.GeometryList by seperator."""
+    splitidx = np.full(len(gl.x_coordinates), False)
+    x, y = gl.x_coordinates.copy(), gl.y_coordinates.copy()
+    idx = np.where(x == by)[0]
+
+    xparts = np.split(x, idx)
+    yparts = np.split(y, idx)
+
+    lists = [
+        mk.GeometryList(xp[min(i, 1) :], yp[min(i, 1) :])
+        for i, (xp, yp) in enumerate(zip(xparts, yparts))
+    ]
+
+    return lists
+
+
+class Mesh2d(BaseModel):
+
+    meshkernel: Optional[mk.MeshKernel] = Field(default_factory=mk.MeshKernel)
+
+    mesh2d_node_x: np.ndarray = np.empty(0, dtype=np.double)
+    mesh2d_node_y: np.ndarray = np.empty(0, dtype=np.double)
+    mesh2d_node_z: np.ndarray = np.empty(0, dtype=np.double)
+
+    mesh2d_edge_x: np.ndarray = np.empty(0, dtype=np.double)
+    mesh2d_edge_y: np.ndarray = np.empty(0, dtype=np.double)
+    mesh2d_edge_z: np.ndarray = np.empty(0, dtype=np.double)
+    mesh2d_edge_nodes: np.ndarray = np.empty((0, 2), dtype=np.int32)
+
+    mesh2d_face_x: np.ndarray = np.empty(0, dtype=np.double)
+    mesh2d_face_y: np.ndarray = np.empty(0, dtype=np.double)
+    mesh2d_face_z: np.ndarray = np.empty(0, dtype=np.double)
+    mesh2d_face_nodes: np.ndarray = np.empty((0, 0), dtype=np.int32)
```

Comment:
I think we could drop the `mesh2d_` prefix, this would also make it easier to define protocols that work on both the 1d and 2d meshes.

Reply:
It is indeed a bit wordy, but i wouldn't remove the prefix. In the 1D mesh the distinction is made to distinguish between mesh1d and network1d. From this perspective, and the consistency with the Ugrid naming (which is followed as much as possible) I would leave the mesh2d for clarity.

Reply:
We will leave it as is for now, and might pick it up if it becomes a requirement in the future.